catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -813,6 +843,179 @@ public void unknownOperation() throws CswException {
         throw new CswException("No such operation", CswConstants.OPERATION_NOT_SUPPORTED, null);
     }
 
+    @Override
+    @DELETE
+    @Path("/subscription/{requestId}")
+    @Produces({MediaType.WILDCARD})
+    public Response deleteRecordsSubscription(@PathParam("requestId") String requestId)
```

Comment:
I believe the main reason I didn't do that initially was that they had a lot of common logic between the two. Now that I have a better understanding Ill see if it makes sense to refactor the code that generates a QueryRequest from a GetRecordsType into a library and then break out the subscription logic.